developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,118 @@
+.. _uuid:
+
+UUID Usage in SOF
+#################
+
+We use universally unique identifier (UUID) for component type
+definition in SOF, and for the whole SOF stack, the UUID usage is
+following below rules:
+
+UUID allocation
+***************
+The UUID of a specific component should be specified in the firmware
+component dirver, e.g. for src component, in sof/src/audio/src.c:
+
+.. code-block:: none
+
+   /* c1c5326d-8390-46b4-aa47-95c3beca6550 */
+   DECLARE_SOF_UUID("src", src_uuid, 0xc1c5326d, 0x8390, 0x46b4,
+                    0xaa, 0x47, 0x95, 0xc3, 0xbe, 0xca, 0x65, 0x50);
+
+UUID in topology
+****************
+We should provide UUID string in topology .m4 file in the exactly same
+format as commented in the firmware, e.g for host it is
+c1c5326d-8390-46b4-aa47-95c3beca6550, so in the topology .m4 file
+tools/topology/m4/src.m4:
+
+.. code-block:: none
+
+   dnl Defines the macro for SRC widget
+   define(SRC_UUID_ARRAY, UUID_ARRAY("c1c5326d-8390-46b4-aa47-95c3beca6550"))
+
+We have a .m4 macro UUID_ARRAY() will help to transfer format in #2 (36
+digits) to 16 Bytes byte array, after compiled, the corresponding ALSA
+UUID token defined here will be generated:
+
+.. code-block:: none
+
+   /* vendor tuple for uuid */
+   struct snd_soc_tplg_vendor_uuid_elem {
+           __le32 token;
+           char uuid[16];
+   } __attribute__((packed));
+
+Linux topology driver
+*********************
+The topology driver will parse the 16Bytes UUID token and sent it to the
+firmware in component creation stage, **for all components**.
+
```

Comment:
Why does the driver care about the UUID ? It should be passing this direct to the FW. The driver only cares about the type atm for IPC1.  In IPC2.0 the driver will only care about type (but with less enums).

Reply:
@lgirdwood are you meaning that not using the generic UUID token, but to create another vendor specific set of tokens for SOF struct sof_uuid (we might need 1 word, 2 short, and 1 string token) ? The driver here is not caring the UUID, it only parsing it and construct the IPC new structure which will be sent for component creation.

Reply:
no, we have to use the full-blown UUID to avoid collisions between vendors.

Reply:
> no, we have to use the full-blown UUID to avoid collisions between vendors.

OK, then paring the 16B UUID token here looks good to me @lgirdwood ?

Reply:
mark this resolved since we are aligned.